includes/data-stores/class-wcs-related-order-store-cached-cpt.php

```diff
@@ -318,16 +318,24 @@ protected function update_related_order_id_cache( $subscription, array $related_
 			$this->update_modified_date_for_related_order_cache( $subscription, $related_order_ids, $current_metadata );
 		}
 
-		// Check if HPOS and data syncing is enabled then manually backfill the related orders cache values to WP Posts table.
-		$this->maybe_backfill_related_order_cache( $subscription, $relation_type, $new_metadata );
-
 		// If there is metadata for this key, update it, otherwise add it.
 		if ( $current_metadata ) {
 			$new_metadata['id'] = $current_metadata->meta_id;
-			return $subscription_data_store->update_meta( $subscription, (object) $new_metadata );
+			$return             = $subscription_data_store->update_meta( $subscription, (object) $new_metadata );
 		} else {
-			return $subscription_data_store->add_meta( $subscription, (object) $new_metadata );
+			$return = $subscription_data_store->add_meta( $subscription, (object) $new_metadata );
 		}
+
+		/**
+		 * Trigger update actions after modifying the subscription's related order cache metadata.
+		 *
+		 * This ensures that functions fired after a subscription update, such as webhooks and those in the DataSynchronizer,
+		 * which sync CPT post data to HPOS tables, are executed.
+		 */
```

Comment:
@mattallan I've just added this comment to explain why we're firing these actions here. I hope you dont mind.

Reply:
> I hope you dont mind.

This has crossed the line 😠 

😄 Thanks for making this suggestion! I would 100% forget why we did this in a years time so it's good to make an inline comment